src/app/(nextjs_migration)/(authenticated)/user/welcome/scanning/page.tsx

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Image from "next/image";
+import Script from "next/script";
+
+import { getL10n } from "../../../../../functions/server/l10n";
+import { getLocale } from "../../../../../../utils/fluent.js";
+
+export async function generateMetadata() {
+  const l10n = getL10n();
+  return {
+    title: "Welcome - Scanning", // FIXME l10n.getString("breach-meta-title"),
+    twitter: {
+      card: "summary_large_image",
+      title: l10n.getString("brand-fx-monitor"),
+      description: l10n.getString("meta-desc-2"),
+      images: ["/images/og-image.webp"],
+    },
+    openGraph: {
+      title: l10n.getString("brand-fx-monitor"),
+      description: l10n.getString("meta-desc-2"),
+      siteName: l10n.getString("brand-fx-monitor"),
+      type: "website",
+      url: process.env.SERVER_URL,
+      images: ["/images/og-image.webp"],
+    },
+  };
+}
+
+export default async function UserWelcomeScanning() {
+  const l10n = getL10n();
+
+  const current = 1;
+  const total = 672;
+  const percentage = ((current / total) * 100).toFixed(1);
+
+  const body = JSON.stringify({
+    first_name: "Test",
+    last_name: "User",
+    addresses: [
+      {
+        state: "NY",
+        city: "New York",
+        zip: "11111",
+        address_line: "1st Ave 1 Apt 1",
+      },
+    ],
+  });
+
+  async function callOneRep(method: string, path: string) {
+    const bearerToken = process.env.ONEREP_API_KEY;
+    const options = {
+      method,
+      headers: {
+        Authorization: `Bearer ${bearerToken}`,
+        "Content-Type": "application/json",
+      },
+    };
+    if (method !== "GET" && method !== "HEAD") {
+      //@ts-ignore FIXME
+      options.body = body;
+    }
+    const result = await fetch(`https://api.onerep.com/${path}`, options);
+    if (!result.ok) {
+      throw new Error("Error connecting to provider");
+    }
+    return result.json();
+  }
+
+  const profile = await callOneRep("POST", "profiles");
+  const scan = await callOneRep("POST", `profiles/${profile.id}/scans`);
+
+  let iterations = 0;
+  const interval = setInterval(async () => {
+    const scanDetails = await callOneRep(
+      "GET",
+      `profiles/${profile.id}/scans/${scan.id}`
```

Comment:
I guess same Q as above re: can `scan` results be `null`|`undefined` from OneRep causing this to error?

Reply:
Actually I do think defining this sort of thing in the type definition would be the most useful thing to do for now, I'll make sure that we're doing the right thing there. As above though, if the scan results aren't present and the user got this far, then something is wrong.

The intended flow is like this:

1. welcome screen
2. user accepts, redirects to details page: user enters information
3. form post to API endpoint: information is validated, onerep profile created and stored in db
4. redirect to scanning page: progress is shown, and scan is initiated and recorded in db
5. redirect to results page: scan result is displayed

At each step, the results are stored in our database so if it's inaccessible that's likely down. We have plans to improve this by introducing queuing (which could fail too of course but e.g. CloudTasks are more distributed and more reliable in general than our primary db, so it will be a step in the right direction).